README.md

```diff
@@ -1,25 +1,39 @@
 # Stairway
-Stairway is a library that provides a framework for running "saga" transactions. 
-The goal is to make a sequence of operations, often over disparate resources,
-run as a transaction; either complete successfully or make no change. A typical
-example in our environment would be an operation where we maintain metadata about
+Stairway is a library that provides a framework for running _saga transactions_. Saga
+transactions, introduced by Hector Garcia-Molina in 1987, use _compensating operations_ to
+rollback the transaction rather than having an service that intermediates activity so is
+able to perform the rollback. The goal is to make a sequence of operations, often over disparate resources,
+run as a transaction; either complete successfully or make no change.
+
+A typical example in our environment is an operation where we maintain metadata about
 an object via a write to a SQL database and create several cloud-platform objects
 to instantiate that object. If any of those operations fail, we want to clean up so
 the state is as if the operation had never occurred.
 
 Stairway provides a way for you to organize your code so that such operations are:
-<ul>
-<li>Atomic - they either happen or don't happen. If something goes wrong, the operation
-undone.</li>
-<li>Recoverable - a failure of the service or system does not lose state. When the
-system recovers, the operation can proceed or rollback.</li>
-</ul>
+ * **Atomic** - they either happen or don't happen. If something goes wrong, the operation undone.
+ * **Recoverable** - a failure of the service or system does not lose state. When the system recovers, the operation can proceed or rollback.
+
+Stairway does not provide as strong a transaction guarantee as a database system. Most database systems are able to
+provide:
+ * **Isolation** -  one transaction runs as if it is the only transaction running
+ * **Consistenty** - a transaction sees a consistent state of the underlying data
+
+Stairway cannot intermediate access to underlying data and objects, so it cannot control
+the view of those objects in the course of running the saga transaction.
 
 With Stairway, you still have to write code with some care to make sure that it results
 in idempotent behavior. The benefit is that Stairway provides a common idiom, and handles
 persistent bookkeeping, retrying, logging, and recovery after failure.
 
+## References
+For more information on developing flights in Stairway see: [Stairway Flight Developer Guide](FLIGHT_DEVELOPER_GUIDE.md)
+
+For information on developing Stairway see [Developing Stairway](DEVELOPMENT.md)
+
 ## Overview
+This section provides an overview of Stairway concepts.
+
```

Comment:
It would be good to have a canonical example to walkthrough that created an artifact and set some permissions on it, e.g. a bucket and its SAM dependencies. It might be more concrete than using a metasyntactic variable like foo.

Reply:
I added a TODO to provide an example.